R/CRS_extracter.R

```diff
@@ -0,0 +1,65 @@
+#' CRS_extracter
+#'
+#' Extracts a coordinate reference system (CRS) from a library of commonly used CRS codes
+#' This also fixes issues with the belge lambert 72 we use.
+#' The EPSG code is 31370, but the proj4string is not the same as the one in the EPSG database.
+#'
+#' @param CRS A character string with the name of the CRS
+#' @param EPSG A logical indicating whether the output should be in EPSG format
+#'
+#' @return A CRS object
+#'
+#' @examples
+#' \dontrun{
+#' # Example of how to use the CRS_extracter function
+#' crs_wgs <- CRS_extracter("WGS", EPSG = FALSE)
+#' crs_bel <- CRS_extracter("BEL72", EPSG = FALSE)
+#'
+#' epsg_wgs <- CRS_extracter("WGS", EPSG = TRUE)
+#'
+#' # Example of how to use the CRS_extracter function in combination with the sf & leaflet packages
+#' library(sf)
+#' library(tidyverse)
+#' library(leaflet)
+#' boswachterijen_df <- boswachterijen$boswachterijen_2024 %>%
+#'   st_transform(crs_bel) %>%
+#'   mutate(centroid = st_centroid(geometry)) %>%
+#'   st_drop_geometry() %>%
+#'   st_as_sf(sf_column_name = "centroid",
+#'            crs = crs_bel) %>%
+#'   st_transform(crs_wgs)
+#'
+#' leaflet() %>%
+#'   addTiles() %>%
+#'   addPolylines(data = boswachterijen$boswachterijen_2024) %>%
+#'   addCircles(data = boswachterijen_df, color = "red", radius = 1000)
+#' }
+#'
+#' @importFrom sp CRS
+#' @importClassesFrom sp CRS
+#' @export
+#' @author Sander Devisscher
+
+CRS_extracter <- function(CRS,
+                          EPSG = TRUE){
+
+  install_sp()
+
+  Lib_CRS <- lib_crs
+
+  if(grepl("wgs", CRS, ignore.case = TRUE)){
+    CRS <- "WGS"
+  }
+
+  if(grepl("bel", CRS, ignore.case = TRUE)){
+    CRS <- "BEL72"
+  }
+
+  if(EPSG == TRUE){
+    CRS_output <- CRS(Lib_CRS$EPSG[CRS == Lib_CRS$CRS_Naam])
+  }else{
```

Comment:
When running this part of the script, I get the following error message:
`Error in CRS(Lib_CRS$EPSG[CRS == Lib_CRS$CRS_Naam]) :  could not find function "CRS". `
I'll have to do the review of the polygon centroid function after this is fixed.

Reply:
I did do devtools::load_all() but it yielded the following error message:
` ℹ The package "osmdata" (>= 0.2.5) is required.
✖ Would you like to install it?

1: Yes
2: No

Selection: 1
Error: 
! error in pak subprocess
Caused by error in get_remote_types(refs):
! Cannot parse package: C:/Users/soria_delva/Documents/GitHub/fistools.
ℹ See ?pkgdepends::pkg_refs() for supported package sources.
Type .Last.error to see the more details. `

Reply:
try `install.packages("osmdata")`

Reply:
It got fixed after I installed the osmdata package independently. There seemed to be a problem when it was installed in the way indicated above

Reply:
And that fixed the error in CRS_extracter 👍

Reply:
yeah 😎

Reply:
function works like a charm 🧙